src/runtime/particle-execution-context.ts

```diff
@@ -57,17 +59,22 @@ export class ParticleExecutionContext implements StorageCommunicationEndpointPro
 
     this.apiPort = new class extends PECInnerPort {
 
-      onDefineHandle(identifier: string, type: Type, name: string) {
+      onDefineHandle(identifier: string, type: Type, name: string, isNG: boolean) {
+        if (isNG) {
+          return new StorageProxyNG(identifier, pec, type);
+        }
         return StorageProxy.newProxy(identifier, type, this, pec, pec.scheduler, name);
       }
 
       onGetBackingStoreCallback(callback: (proxy: StorageProxy, key: string) => void, type: Type, name: string, id: string, storageKey: string) {
+        // TODO create NG StorageProxy.
```

Comment:
You have the necessary info here to create the StorageProxyNG as well. If you just check Flags.useNewStorageStack, you should be able to construct the correct type here too.

Reply:
Done